flowetl/flowetl/flowetl/operators/staging/sql/create_and_fill_day_sightings_table.sql

```diff
@@ -0,0 +1,54 @@
+BEGIN;
+
+-- TODO: Move to flowdb
+CREATE SCHEMA IF NOT EXISTS reduced;
+
+DROP TABLE IF EXISTS reduced.sightings_{{ds_nodash}};
+
+--TODO: msisdn and locationid to bytea from text
+CREATE TABLE reduced.sightings_{{ds_nodash}}(
+ LIKE reduced.sightings,
+ CONSTRAINT sightings_{{ds_nodash}}_check
+    CHECK (sighting_date >= DATE '{{ds_nodash}}' AND sighting_date < DATE '{{ds_nodash}}' + 1)
+);
+
+WITH ranked AS (
+    SELECT date_time, cell_id, msisdn, event_type,
+           dense_rank() OVER ( PARTITION BY msisdn, cell_id
+                               ORDER BY date_time
+                             ) AS cell_id_rank,
+           dense_rank() OVER ( PARTITION BY msisdn
+                               ORDER BY date_time
+                             ) AS date_rank
+    FROM staging_table_{{ds_nodash}}
+    ORDER BY date_time
+), group_ranked AS (
+    SELECT date_time, cell_id, msisdn, cell_id_rank, date_rank,
+           event_type, cell_id_rank - date_rank AS group_rank
+    FROM ranked
+    ORDER BY date_time
+), aggregated AS (
+    SELECT msisdn,
+           row_number() OVER ( PARTITION BY msisdn
+                               ORDER BY min(date_time), max(date_time)
+                             ) AS row_id,
+           cell_id,
+           array_agg(date_time ORDER BY date_time) AS cons_dates,
+           array_agg(event_type ORDER BY date_time) AS cons_events
+    FROM group_ranked
+    GROUP BY msisdn, cell_id, group_rank
+    ORDER BY msisdn, row_id
+)
+INSERT INTO reduced.sightings_{{ds_nodash}}
```

Comment:
I _think_ the issue here is that this will trigger WAL, which can be skipped if using a `CREATE TABLE AS ...`.

Reply:
I think that'd lose us the constraint-at-table-creation, which is needed for the partitioning. Maybe we can disable the WAL with `CREATE UNLOGGED TABLE`, given this is only a transient table created as part of the DAG?